lib/evmone/eof.cpp

```diff
@@ -13,6 +13,92 @@ namespace
 {
 constexpr uint8_t FORMAT = 0xef;
 constexpr uint8_t MAGIC = 0x00;
+constexpr uint8_t TERMINATOR = 0x00;
+constexpr uint8_t CODE_SECTION = 0x01;
+constexpr uint8_t DATA_SECTION = 0x02;
+
+std::pair<EOF1Header, EOFValidationErrror> validate_eof_headers(bytes_view code) noexcept
+{
+    enum class State
+    {
+        section_id,
+        section_size,
+        terminated
+    };
+
+    auto state = State::section_id;
+    uint8_t section_id = 0;
+    size_t section_sizes[3] = {0, 0, 0};
```

Comment:
What are the 3 sizes?

Reply:
Sizes for each possible value of section_id, 0th one is always 0 for terminator.

In later commit this is replaces with `std::array<uint16_t, MAX_SECTION + 1>`